src/sonic-host-services/scripts/determine-reboot-cause

```diff
@@ -98,29 +98,35 @@ def find_proc_cmdline_reboot_cause():
 
     return proc_cmdline_reboot_cause
 
+
 def get_reboot_cause_from_platform():
     # Until all platform vendors have provided sonic_platform packages,
     # if there is no sonic_platform package installed, we only provide
     # software-related reboot causes.
+    hardware_reboot_cause_major = "N/A"
+    hardware_reboot_cause_minor = "N/A"
```

Comment:
I think it might be best to move the definitions on lines 122-123 to the top of the file. E.g., 

```python
    REBOOT_CAUSE_HARDWARE_OTHER = "Hardware - Other"
    REBOOT_CAUSE_NON_HARDWARE = "Non-Hardware"
```

And get rid of `REBOOT_CAUSE_HARDWARE_NA`

Then here, set the default values as follows:

```python
    hardware_reboot_cause_major = REBOOT_CAUSE_NON_HARDWARE
    hardware_reboot_cause_minor = "N/A"
```

Then revert your change on line 129. As long as we are logging a warning message, we will have a breadcrumb trail to understand that we were unable to obtain a hardware reboot cause.

Reply:
I took the liberty to do a little more than that - combined two functions for hardware reboot cause handling into one and removed extra variables. Please re-review.